From ab7da0672172e5d7def87c05d58d76a5970c7ac7 Mon Sep 17 00:00:00 2001
From: Lokesh Mandvekar <lsm5@fedoraproject.org>
Date: Mon, 9 Mar 2015 12:47:24 -0400
Subject: [PATCH] replace closed with _fp.isclosed() for rhel7

Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>
---
 docker/client.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/docker/client.py b/docker/client.py
index 79726aa..5a5a97c 100644
--- a/docker/client.py
+++ b/docker/client.py
@@ -185,7 +185,7 @@ class Client(requests.Session):
         """Generator for data coming from a chunked-encoded HTTP response."""
         if response.raw._fp.chunked:
             reader = response.raw
-            while not reader.closed:
+            while not reader._fp.isclosed():
                 # this read call will block until we get a chunk
                 data = reader.read(1)
                 if not data:
--
1.8.3.1

