diff -up docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/api/client/commands.go.signals docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/api/client/commands.go
--- docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/api/client/commands.go.signals	2014-07-18 14:59:08.000000000 -0400
+++ docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/api/client/commands.go	2014-07-18 16:04:50.376517013 -0400
@@ -585,7 +585,7 @@ func (cli *DockerCli) CmdRestart(args ..
 }
 
 func (cli *DockerCli) forwardAllSignals(cid string) chan os.Signal {
-	sigc := make(chan os.Signal, 1)
+	sigc := make(chan os.Signal, 128)
 	signal.CatchAll(sigc)
 	go func() {
 		for s := range sigc {
