diff -up docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/archive.go.remove-vendored-tar docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/archive.go
--- docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/archive.go.remove-vendored-tar	2014-07-18 14:59:08.000000000 -0400
+++ docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/archive.go	2014-07-18 15:38:07.402830750 -0400
@@ -18,7 +18,7 @@ import (
 
 	"github.com/dotcloud/docker/pkg/system"
 	"github.com/dotcloud/docker/utils"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 )
 
 type (
diff -up docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/archive_test.go.remove-vendored-tar docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/archive_test.go
--- docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/archive_test.go.remove-vendored-tar	2014-07-18 14:59:08.000000000 -0400
+++ docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/archive_test.go	2014-07-18 15:38:07.402830750 -0400
@@ -11,7 +11,7 @@ import (
 	"testing"
 	"time"
 
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 )
 
 func TestCmdStreamLargeStderr(t *testing.T) {
diff -up docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/changes.go.remove-vendored-tar docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/changes.go
--- docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/changes.go.remove-vendored-tar	2014-07-18 14:59:08.000000000 -0400
+++ docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/changes.go	2014-07-18 15:38:07.402830750 -0400
@@ -12,7 +12,7 @@ import (
 
 	"github.com/dotcloud/docker/pkg/system"
 	"github.com/dotcloud/docker/utils"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 )
 
 type ChangeType int
diff -up docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/diff.go.remove-vendored-tar docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/diff.go
--- docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/diff.go.remove-vendored-tar	2014-07-18 14:59:08.000000000 -0400
+++ docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/diff.go	2014-07-18 15:38:07.403830752 -0400
@@ -9,7 +9,7 @@ import (
 	"strings"
 	"syscall"
 
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 )
 
 // Linux device nodes are a bit weird due to backwards compat with 16 bit device nodes.
diff -up docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/wrap.go.remove-vendored-tar docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/wrap.go
--- docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/wrap.go.remove-vendored-tar	2014-07-18 14:59:08.000000000 -0400
+++ docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/archive/wrap.go	2014-07-18 15:38:07.403830752 -0400
@@ -2,7 +2,7 @@ package archive
 
 import (
 	"bytes"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 	"io/ioutil"
 )
 
diff -up docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/graph/tags_unit_test.go.remove-vendored-tar docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/graph/tags_unit_test.go
--- docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/graph/tags_unit_test.go.remove-vendored-tar	2014-07-18 14:59:08.000000000 -0400
+++ docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/graph/tags_unit_test.go	2014-07-18 15:38:07.403830752 -0400
@@ -6,7 +6,7 @@ import (
 	_ "github.com/dotcloud/docker/daemon/graphdriver/vfs" // import the vfs driver so it is used in the tests
 	"github.com/dotcloud/docker/image"
 	"github.com/dotcloud/docker/utils"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 	"io"
 	"os"
 	"path"
diff -up docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/integration/api_test.go.remove-vendored-tar docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/integration/api_test.go
--- docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/integration/api_test.go.remove-vendored-tar	2014-07-18 15:38:07.403830752 -0400
+++ docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/integration/api_test.go	2014-07-18 15:39:42.543049533 -0400
@@ -13,12 +13,12 @@ import (
 	"testing"
 	"time"
 
+	"archive/tar"
 	"github.com/dotcloud/docker/api"
 	"github.com/dotcloud/docker/api/server"
 	"github.com/dotcloud/docker/engine"
 	"github.com/dotcloud/docker/image"
 	"github.com/dotcloud/docker/runconfig"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
 )
 
 func TestGetImagesJSON(t *testing.T) {
diff -up docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/integration/utils_test.go.remove-vendored-tar docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/integration/utils_test.go
--- docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/integration/utils_test.go.remove-vendored-tar	2014-07-18 14:59:08.000000000 -0400
+++ docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/integration/utils_test.go	2014-07-18 15:38:07.404830754 -0400
@@ -13,7 +13,7 @@ import (
 	"testing"
 	"time"
 
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 
 	"github.com/dotcloud/docker/builtins"
 	"github.com/dotcloud/docker/daemon"
diff -up docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/utils/tarsum.go.remove-vendored-tar docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/utils/tarsum.go
--- docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/utils/tarsum.go.remove-vendored-tar	2014-07-18 14:59:08.000000000 -0400
+++ docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/utils/tarsum.go	2014-07-18 15:38:07.404830754 -0400
@@ -5,7 +5,7 @@ import (
 	"compress/gzip"
 	"crypto/sha256"
 	"encoding/hex"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 	"hash"
 	"io"
 	"sort"
diff -up docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/utils/tarsum_test.go.remove-vendored-tar docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/utils/tarsum_test.go
--- docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/utils/tarsum_test.go.remove-vendored-tar	2014-07-18 15:38:07.404830754 -0400
+++ docker-2abf9ee3340c831b17e9df16b8650ee6c8788f6c/utils/tarsum_test.go	2014-07-18 15:41:20.405274575 -0400
@@ -1,6 +1,7 @@
 package utils
 
 import (
+	"archive/tar"
 	"bytes"
 	"crypto/rand"
 	"fmt"
@@ -8,8 +9,6 @@ import (
 	"io/ioutil"
 	"os"
 	"testing"
-
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
 )
 
 type testLayer struct {
